src/excalidraw-app/data/firebase.ts

```diff
@@ -145,55 +141,76 @@ export const isSavedToFirebase = (
   return true;
 };
 
+const getDataURLMimeType = (dataURL: DataURL): string => {
+  return dataURL.split(",")[0].split(":")[1].split(";")[0];
```

Comment:
You probably don't want to do that. This string is multiple megabytes. You don't want to use split on this which is going to search through the entire string and allocates a bunch of sub-strings.

I'd recommend using indexOf(',') and then extracting the beginning via substr. Then you can do the normal split magic.

Reply:
A quick hack I used from an earlier quick hack and this thing will likely get removed before merging. But a good point. A regex would likely be even better.